book/src/dev/mempool-specification.md

```diff
@@ -0,0 +1,206 @@
+# Mempool Specification
+
+The Zebra mempool handles unmined Zcash transactions: collecting them from peers, verifying them, storing them in memory, providing APIs for other components to access them, and gossiping transactions to peers. This document specifies the architecture, behavior, and interfaces of the mempool.
+
+## Overview
+
+The mempool is a fundamental component of the Zebra node, responsible for managing the lifecycle of unmined transactions. It provides an in-memory storage for valid transactions that haven't yet been included in a block, and offers interfaces for other components to interact with these transactions.
+
+Key responsibilities of the mempool include:
+- Accepting new transactions from the network
+- Verifying transactions against consensus rules
```

Comment:
```suggestion
- Verifying transactions against a subset of consensus rules
```

I think the mempool can't verify all consensus rules related to transactions but just some of them.